src/compute/src/compute_state.rs

```diff
@@ -844,6 +861,17 @@ impl PendingPeek {
             }
         }
 
-        Ok(results)
+        let count_size = std::mem::size_of::<NonZeroUsize>();
+        if results.iter().fold(0, |acc: usize, (row, _count)| {
+            acc.saturating_add(row.byte_len())
+                .saturating_add(count_size)
+        }) < usize::cast_from(max_result_size)
+        {
+            Ok(results)
+        } else {
+            Err(format!(
+                "result exceeds max size of {max_result_size} bytes"
+            ))
+        }
```

Comment:
Just realized that it's probably better to track this as we accumulate rows and not wait until the very end. I'm going to make that change now.